### PR TITLE
build!: remove neovim qt

### LINKED ISSUE #21209: Do not put common DLLs in the same directory as the executable
### Feature already in Vim?

_No response_

### Feature description

I recently tried to upgrade Neovim on Windows (via Chocolatey), but the upgrade script was unable to delete some libraries, including libEGL.dll, libGLESv2.dll, and some others. It turned out that these were being kept open by another application.

My guess is that because nvim.exe is in the system path, the other application was trying to use similar libraries (my suspicion is because it was using Qt5), found these libraries on the system path, and held them open. While the other application is the main culprit (I've filed a bug with them as well), keeping local copies of these DLLs in a different directory off the system path is probably wise.

### PR DESCRIPTION
**Pull request by dundargoc**
Neovim QT was originally bundled on Windows as a response to the then
lackluster terminal options. The situation has dramatically changed,
with viable options such as Windows terminal, Alacritty and Wezterm to
name a few. The Windows build no longer needs this special treatment for
neovim to be usable.

Pros:
  - Release builds will be smaller.
  - Less maintenance burden.
  - Clearer separation of responsibility (neovim issues go to the neovim
    repo and neovim-qt issues to the neovim-qt repo).
  - More consistent treatment between platforms.

Cons:
  - Users who've come to expect neovim-qt to be bundled with nvim will
    need to adjust and download neovim-qt from
    https://github.com/equalsraf/neovim-qt instead.
  - Similarly, build scripts will need to be adjusted to reflect this
    change.

Closes https://github.com/neovim/neovim/issues/21209.

### CODE REVIEW

File: .github/workflows/notes.md
Comment (zeertzjq) [inline .github/workflows/notes.md:19]:
```suggestion
3. Search and run `nvim.exe` on your CLI of choice
```

File: .github/workflows/notes.md
Comment (zeertzjq) [inline .github/workflows/notes.md:19]:
```suggestion
3. Run `nvim.exe` on your CLI of choice
```

File: .github/workflows/notes.md
Comment (zeertzjq) [inline .github/workflows/notes.md:13]:
```suggestion
3. Run `nvim.exe` on your CLI of choice
```

File: runtime/doc/news.txt
Comment (clason) [inline runtime/doc/news.txt:84]: I don't think this needs a news entry; this is more a project-level change that doesn't affect users _already having Neovim_.

Marking this as breaking is enough to put it front and center in the release notes.

**Comment by dundargoc** [comment]
Pinging @equalsraf for comments and awareness as they're responsible for nvim qt as I understand it.

**Comment by justinmk** [comment]
I recall that it was discussed earlier in the year. If there was a conclusion from that discussion (cost/benefit analysis) it should be included in the commit message / description here.

One benefit is that the release archive will be much smaller.

I don't perceive a big maintenance burden here, but corrections welcome.

**Comment by clason** [comment]
Maintenance burden includes regular issues on this repo about neovim-qt problems, which wastes time for both maintainers and users (getting the run-around).

**Comment by dundargoc** [comment]
The previous discussion on this is [here](https://github.com/neovim/neovim/pull/21866). I can't see any cost/benefit analysis there. @justinmk's [comment](https://github.com/neovim/neovim/pull/21866#issuecomment-1400109363) is the closest thing to a conclusion I can see, but I don't know what "Once we have better messaging and discovery of available GUIs on [neovim.io](https://neovim.io/) , then we could consider dropping nvim-qt in our releases." means in practice.

I'm unsure how to make progress on this or what I need to do to unblock it.

**Comment by bfredl** [comment]
That sounds like a statement made before TUI got generally usable on windows.

The generally discoverable GUI on windows is now the TUI, just like on POSIX plattforms. So I would say we have reached that condition.

**Comment by clason** [comment]
Several Neovim GUIs are available from scoop (extras): https://scoop.sh/#/apps?q=neovim

So maybe just add a line at the end of https://github.com/neovim/neovim/wiki/Installing-Neovim#scoop to that effect?

**Comment by dundargoc** [comment]
> So maybe just add a line at the end of https://github.com/neovim/neovim/wiki/Installing-Neovim#scoop to that effect?

If that'll help move this PR forward then sure, although I don't personally understand the reasoning. The premise of this PR is that Windows is now a big boy OS with features like terminals and package managers, and no longer needs special treatment.

**Review by clason — APPROVED** [review]

**Comment by equalsraf** [comment]
> Pinging @equalsraf for comments and awareness as they're responsible for nvim qt as I understand it.

Thanks for the ping lads. I happily endorse whatever you consider is best here.

> That sounds like a statement made before TUI got generally usable on windows.

This, I think is the only aspect that matters. If the TUI is working well on windows then task complete.

It is probably worth putting some instructions in the wiki on how to start nvim.exe in the windows terminal, or other similar tidbits. But I'm sure those will emerge in time.

**Comment by dundargoc** [comment]
@equalsraf glad to hear :)

Do you need to make any preparations before we pull the plug on nvim-qt? I'm ready to accommodate and help this process in any way I can.

**Comment by equalsraf** [comment]
> Do you need to make any preparations before we pull the plug on nvim-qt? I'm ready to accommodate and help this process in any way I can.

We are good on my end. As far as I can see there is also a neovim-qt pkg in scoop.

**Review by justinmk — APPROVED** [review]
scoop/winget address my main concern. Updating our website to better feature high-quality GUIs, doesn't need to block this.

**Comment by dundargoc** [comment]
@justinmk I'm probably biased, but I can't come up with any real cons for this.

**Comment by justinmk** [comment]
The con is mostly inertia / user expectations + "sunk cost" in terms of build scripting. But we have a path forward for discovering GUIs (website, maybe more TBD), and the Windows-only nature of this current approach is not good. So we should move forward.

**Comment by crash5** [comment]
strictly speaking maybe it is not a breaking change but it would be really helpful to state the neovim-qt removal in the changelog

**Pull request merged**
